### PR TITLE
用creative.is_enabled_for替换minetest.setting_getbool("creative_mode")

### PR DESCRIPTION
**Pull request by zaoqi**

### CODE REVIEW

File: mods/creative/init.lua [hunk 1]
```diff
@@ -8,35 +8,33 @@ end
...
+		groupcaps = {
+			crumbly = caps,
+			cracky  = caps,
+			snappy  = caps,
+			choppy  = caps,
+			oddly_breakable_by_hand = caps,
+		},
+		damage_groups = {fleshy = 10},
+	}
+})
```
Comment (sfan5) [inline mods/creative/init.lua]: This would override the hand for *all* players and should be kept for now.

File: mods/doors/init.lua [hunk 2]
```diff
@@ -587,7 +587,7 @@ function doors.register_trapdoor(name, def)
 			meta:set_string("doors_owner", pn)
 			meta:set_string("infotext", "Owned by "..pn)
 
-			return minetest.setting_getbool("creative_mode")
+			return creative.is_enabled_for(name)
```
Comment (sfan5) [inline mods/doors/init.lua]: should be `creative.is_enabled_for(pn)`

**Comment by paramat** [comment]
^ What SmallJoker wrote.

**Comment by rubenwardy** [comment]
Uncertain now - it' probably worth not making this a hard dep

**Comment by paramat** [comment]
@rubenwardy what's your position? do you consider the PR is needed but the dependancies should either be removed or made soft?

**Comment by zaoqi** [comment]
@rubenwardy

File: mods/beds/api.lua [hunk 3]
```diff
@@ -89,7 +89,7 @@ function beds.register_bed(name, def)
 			minetest.set_node(pos, {name = name .. "_bottom", param2 = dir})
 			minetest.set_node(botpos, {name = name .. "_top", param2 = dir})
 
-			if not minetest.setting_getbool("creative_mode") then
+			if not (creative and creative.is_enabled_for and creative.is_enabled_for(placer:get_player_name())) then
```
Comment (SmallJoker) [inline mods/beds/api.lua]: Line**s** too long, pelase break them up into two

**Comment by zaoqi** [comment]
@rubenwardy @SmallJoker

File: mods/fire/init.lua [hunk 4]
```diff
@@ -103,7 +103,8 @@ minetest.register_tool("fire:flint_and_steel", {
 				minetest.set_node(pointed_thing.above, {name = "fire:basic_flame"})
 			end
 		end
-		if not minetest.setting_getbool("creative_mode") then
+		if not (creative and creative.is_enabled_for
+				and creative.is_enabled_for(player_name)) then
```
Comment (SmallJoker) [inline mods/fire/init.lua]: `player_name` is not defined here.

File: mods/doors/init.lua [hunk 5]
```diff
@@ -322,7 +322,8 @@ function doors.register(name, def)
 				meta:set_string("infotext", "Owned by " .. pn)
 			end
 
-			if not minetest.setting_getbool("creative_mode") then
+			if not (creative and creative.is_enabled_for
+					and creative.is_enabled_for(placer:get_player_name())) then
```
Comment (SmallJoker) [inline mods/doors/init.lua]: `creative.is_enabled_for(pn))`

**Comment by zaoqi** [comment]
@SmallJoker

**Comment by zaoqi** [comment]
@sfan5

**Comment by SmallJoker** [comment]
Merged as 077316b

**Pull request closed without merging**